js/components/tabs.js

```diff
@@ -27,6 +27,25 @@ class Tabs extends ImmutableComponent {
     this.onNextPage = this.onNextPage.bind(this)
     this.onNewTabLongPress = this.onNewTabLongPress.bind(this)
     this.wasNewTabClicked = this.wasNewTabClicked.bind(this)
+    this.onMouseLeave = this.onMouseLeave.bind(this)
+    this.onFrameClose = this.onFrameClose.bind(this)
+
+    this.state = {
+      fixTabWidth: null
+    }
+  }
+
+  onMouseLeave () {
+    this.setState({
+      fixTabWidth: null
+    })
+    this.forceUpdate()
```

Comment:
@gyandeeps the ideal approach would be flux-like and to do the following:
- store the state in our AppStore / AppState. You can [view the current layout of the appState here](https://github.com/brave/browser-laptop/blob/master/docs/state.md)
- Inside `onFrameClose`, create and call a new [appAction](https://github.com/brave/browser-laptop/blob/master/js/actions/appActions.js) which has should also have an event based name.  `onFrameClose` seems OK
- The action then (like other actions) makes a call to dispatcher. This gets handled in [appStore.js](https://github.com/brave/browser-laptop/blob/master/js/stores/appStore.js#L376)
- AppStore then updates the application state
- `main.js` should be binding the appState to this control. [Here's an example with a different UI control](https://github.com/brave/browser-laptop/blob/master/js/components/main.js#L1169)

Using the immutable control approach is ideal because components themself don't have state; everything goes through the app store.  Even if we insisted on keeping state, it would be better to just change the control to inherit from `React.Component`